6-manifest-preview-capability/README.md

```diff
@@ -0,0 +1,370 @@
+- Does this proposal involve coordinating behavior between components?
+  - (i.e. the Zarf Agent and CLI? The init package and the CLI?)
+-->
+
+N/A
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a ZEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling acceptance of the ZEP
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the ZEP was available
+- the version of Kubernetes where the ZEP graduated to general availability
+- when the ZEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this ZEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+One alternative was to use `zarf dev preview`, the thought being that this command will be run by creators while developing a package. However, given the similarities between this command and `zarf package inspect` we decided that it made for a more cohesive user experience to have both commands under the same parent.
+
+I pondered if we should accept a component flag as we do on deploy during package inspect or package preview. This would allow people to get a more accurate view of the manifests that they intend to deploy if they are using optional components. However, this flag would only make sense alongside the  It does make the code more complicated, but it can give a better view of what will actually be deployed. Given the added complexity, we are not added this for now, but it could be an enhancement in the future. 
+
+Another alternative would be to have completely separate commands that show the zarf.yaml file and the commands that show the manifests and values files. The `zarf package preview` command would exist solely to view a zarf.yaml and the `zarf package inspect` command would be unchanged. Instead we'd introduce the commands `zarf dev show-manifests` and `zarf dev show-values-files`. These commands would have the `--deploy-set` and `--create-set` flags, while `zarf package preview` would only need `--set`. This would increase the surface area of the CLI, but give commands a more distinct purpose. This would help the flags be more sensible as well. Helm templating would require a `--kube-version` for charts with a certain version set. A `--components` flag would be nice for users who want to see the manifests they are deploying without optional components included, but wouldn't make sense for the `zarf package preview` command since all components are included on create regardless of if they are used or not. The `show-manifests` and `show-values-files` commands would take either a zarf.yaml or a zarf package.
```

Comment:
Imo the `dev` command branch is a bit of an antipattern. If functionality is important enough to be included in the CLI it should be first class, and if it doesn't fit existing commands enough to be a subcommand it should have its own subtree (e.g `zarf inspect ...xyz` or `zarf show ...abc). `package preview` + flags here works fine

Reply:
I wouldn't say `dev` is an anti-pattern, the intent around dev is for commands for those "developing" a Zarf package. For example, `dev lint` and `dev find-images`. I could see the argument that since these `show` commands might be used by both creators and deployers they deserve a separate bucket. I'd still lean towards `dev` though as I feel it's more important for creators, and being in the `dev` root would lead to better discoverability. 

Perhaps we could introduce two sets of commands `zarf dev show-manifests` / `zarf dev show-values-files` and `zarf package show-manifests` / `zarf package show-values-files`. The `dev` commands would run in a package directory while the `package` commands would run on a tar file. This would keep `dev` consistent with other commands, but does come at a cost of increasing surface area.

Reply:
separated out commands